x/ccv/democracy/distribution/module.go

```diff
@@ -0,0 +1,132 @@
+package distribution
+
+import (
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/types/module"
+
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	distr "github.com/cosmos/cosmos-sdk/x/distribution"
+	"github.com/cosmos/cosmos-sdk/x/distribution/keeper"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	consumertypes "github.com/cosmos/interchain-security/x/ccv/consumer/types"
+
+	distrtypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+var (
+	_ module.AppModule           = AppModule{}
+	_ module.AppModuleBasic      = AppModuleBasic{}
+	_ module.AppModuleSimulation = AppModule{}
+)
+
+// AppModule embeds the Cosmos SDK's x/staking AppModuleBasic.
+type AppModuleBasic struct {
+	distr.AppModuleBasic
+}
+
+// AppModule embeds the Cosmos SDK's x/distribution AppModule
+type AppModule struct {
+	// embed the Cosmos SDK's x/distribution AppModule
+	distr.AppModule
+
+	keeper        keeper.Keeper
+	accountKeeper distrtypes.AccountKeeper
+	bankKeeper    distrtypes.BankKeeper
+	stakingKeeper stakingkeeper.Keeper
+
+	feeCollectorName string
+}
+
+// NewAppModule creates a new AppModule object using the native x/distribution module
+// AppModule constructor.
+func NewAppModule(
+	cdc codec.Codec, keeper keeper.Keeper, ak distrtypes.AccountKeeper,
+	bk distrtypes.BankKeeper, sk stakingkeeper.Keeper, feeCollectorName string,
+) AppModule {
+	distrAppMod := distr.NewAppModule(cdc, keeper, ak, bk, sk)
+	return AppModule{
+		AppModule:        distrAppMod,
+		keeper:           keeper,
+		accountKeeper:    ak,
+		bankKeeper:       bk,
+		stakingKeeper:    sk,
+		feeCollectorName: feeCollectorName,
+	}
+}
+
+// BeginBlocker mirror functionality of cosmos-sdk/distribution BeginBlocker
+// however it allocates no proposer reward
+func (am AppModule) BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock) {
+	defer telemetry.ModuleMeasureSince(distrtypes.ModuleName, time.Now(), telemetry.MetricKeyBeginBlocker)
+
+	// TODO this is Tendermint-dependent
+	// ref https://github.com/cosmos/cosmos-sdk/issues/3095
+	if ctx.BlockHeight() > 1 {
+		am.AllocateTokens(ctx)
+	}
+
+	// record the proposer for when we payout on the next block
+	consAddr := sdk.ConsAddress(req.Header.ProposerAddress)
+	am.keeper.SetPreviousProposerConsAddr(ctx, consAddr)
```

Comment:
Is this used for anything any more?

Reply:
It can be safely removed since it is not used in democracy distribution module. In original core democracy module, it is used only in AllocateTokens function to calculate the rewards based on proposers but in overridden democracy distribution module,  AllocateTokens does not use it since the logic for rewards distribution is based on representatives and its powers. Also, it is used in distribution keeper ExportGenesis  function which export this key and in the case of democracy consumer chain this key will be empty (export does not fail, it is successfully done)